lib/dts-lib.robot

```diff
@@ -0,0 +1,130 @@
+*** Settings ***
+Library     ../robot-venv/lib/python3.11/site-packages/robot/libraries/Collections.py
+
+
+*** Keywords ***
+Enable Network Boot
+    Power On
+    ${setup_menu}=    Enter Setup Menu Tianocore And Return Construction
+    ${dasharo_menu}=    Enter Dasharo System Features    ${setup_menu}
+    ${nw_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Networking Options
+    Set Option State    ${nw_menu}    Enable network boot    ${TRUE}
+    Save Changes And Reset    2    4
+
+Boot Dasharo Tools Suite
+    [Documentation]    Keyword allows to boot Dasharo Tools Suite. Takes the
+    ...    boot method (from USB or from iPXE) as parameter.
+    [Arguments]    ${dts_booting_method}
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    IF    '${dts_booting_method}'=='USB'
+        Enter Submenu From Snapshot    ${boot_menu}    USB SanDisk 3.2Gen1
+    ELSE IF    '${dts_booting_method}'=='USB_emulated'
+        Enter Submenu From Snapshot    ${boot_menu}    PiKVM Composite KVM Device
+        Sleep    1s
+        Press Key N Times    1    ${ENTER}
+    ELSE IF    '${dts_booting_method}'=='iPXE'
+        Enter Submenu From Snapshot    ${boot_menu}    iPXE Network Boot
+        ${ipxe_menu}=    Get Submenu Construction    iPXE Shell
+        Enter Submenu From Snapshot    ${ipxe_menu}    Dasharo Tools Suite
+    ELSE
+        FAIL    Unknown or improper connection method: ${dts_booting_method}
+    END
+    Read From Terminal Until    Enter an option:
+
+Check HCL Report Creation
+    [Documentation]    Keyword allows to check if the Dasharo Tools Suite
+    ...    option for creating HCL report works correctly.
+    Enter Shell In DTS
+    Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
```

Comment:
Again I see `    Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH` just after ` Enter Shell In DTS` so maybe it should be a part of it.

Also, isn't it only required for MSI/PiKVM (where telnet can only read, but not write)?

Reply:
Every occurrence moved from the test file to the keyword